server/chat.js

```diff
@@ -220,17 +221,13 @@ module.exports = function(imports) {
         let chat = yield Chat.findOne({
             _id: req.params.chatId,
         });
-        if(
-            (chat.isTwoPeople 
-                && (chat.audience.users[0].toString() == req.user._id.toString() 
-                || chat.audience.users[1].toString()) == req.user._id.toString()) 
+        if(isUserInAudience(req.user, chat.audience)
+            && (chat.isTwoPeople  
             || util.positions.isUserAdmin(req.user) 
-            || req.user._id.toString() === chat.creator.toString()
+            || req.user._id.toString() === chat.creator.toString())
         ){
-            
             yield chat.remove(); 
-        }
-        else {
+        } else {
             res.status(403).end("You do not have permission");
         }
         res.end();
```

Comment:
`res.end` will be called twice if the user does not have permission. Never call `res.end` twice.